imap_processing/tests/codice/test_codice_l1a.py

```diff
@@ -228,17 +228,11 @@ def test_l1a_validate_data_arrays(test_l1a_data: xr.Dataset, index):
         validation_dataset = load_cdf(VALIDATION_DATA[index])
 
         for counter in counters:
-            # Ensure the data array shapes are equal
-            assert (
-                processed_dataset[counter].data.shape
-                == validation_dataset[counter].data.shape
+            # Ensure the data arrays are equal
+            np.testing.assert_equal(
+                processed_dataset[counter].data, validation_dataset[counter].data
```

Comment:
Give the xarray version a try, and I think you can drop the `.data` access as well.
```suggestion
            xr.testing.assert_equal(
                processed_dataset[counter], validation_dataset[counter]
```

Reply:
Agreed, `np.testing` is the way to go here then!